test/VaultOUSD.ts

```diff
@@ -5,85 +5,124 @@ import {
     addressOUSD,
     abiOUSDToken,
     helperSwapETHWithOUSD,
+    defaultBlockNumber,
 } from "./MainnetHelper";
+import { buildContractTestContext, ContractTestContext } from "./ContractTestContext";
 
 const getDecimal = (naturalNumber) => {
     return ethers.utils.parseEther(naturalNumber.toString());
 };
 
+const getFloatFromBigNum = (bigNumValue) => {
+    return parseFloat(ethers.utils.formatEther(bigNumValue));
+};
+
 describe("VaultOUSD test suit", function () {
-    let tokenVault;
-    let tokenOUSD;
     let owner;
     let addr1;
     let addr2;
+    let addr3;
     let sharesOwnerAddress;
+    let r: ContractTestContext;
 
     const addr1Deposit = 10;
     const addr2Deposit = 20;
     const interestIntoVault = 10;
-    before(async function () {
-        [owner, addr1, addr2] = await ethers.getSigners();
-        helperResetNetwork(14533286);
-        tokenOUSD = new ethers.Contract(addressOUSD, abiOUSDToken, owner);
-        const contractVault = await ethers.getContractFactory("VaultOUSD");
-        tokenVault = await contractVault.deploy(tokenOUSD.address, "VaultOUSD", "VOUSD");
+
+    async function setupAndResetState () {
```

Comment:
can this be in "beforeEach" ?

Reply:
These tests are built in a way that I dont want to reset the state between each `it`. Having the reset beforeEach resets the network on each test. Could not find a way to run the setup just before each describe